pychunkedgraph/app/app_utils.py

```diff
@@ -206,13 +192,35 @@ def get_cg(table_id, skip_cache: bool = False):
     )
     formatter.converter = gmtime
     handler.setFormatter(formatter)
-
     logger.addHandler(handler)
 
-    # Create ChunkedGraph
+
+def ensure_correct_version(cg: ChunkedGraph) -> bool:
+    version = cg.client.read_graph_version()
+    current_major_version = int(__version__.split(".")[0])
+    try:
+        graph_major_version = int(version.split(".")[0])
+        return graph_major_version == current_major_version
+    except (AttributeError, TypeError):
+        return False
+
+
+def get_cg(table_id, skip_cache: bool = False):
+    current_app.table_id = table_id
+    if skip_cache is False:
+        try:
+            return CACHE[table_id]
+        except KeyError:
+            pass
+
+    setup_logger(table_id)
     cg = ChunkedGraph(graph_id=table_id, client_info=get_default_client_info())
     if skip_cache is False:
         CACHE[table_id] = cg
+
+    version_validity = ensure_correct_version(cg)
+    whitelisted = table_id in current_app.config["PCG_GRAPH_IDS"]
+    assert version_validity or whitelisted, f"Invalid graph name {table_id}."
```

Comment:
version_validity should take `precedence` over `whitelisted`. If there is a version mismatch, an error should be raised. If no version is stored in the PCG, it should fall back to the whitelist. It would be great to raise a warning in the latter case